core/arch/arm/kernel/boot.c

```diff
@@ -1133,6 +1142,138 @@ static int mark_tzdram_as_reserved(struct dt_descriptor *dt)
 				   CFG_TZDRAM_SIZE);
 }
 
+#ifdef CFG_NSEC_DTB_OVERLAY_ADDR
+static int add_descriptor_node(struct dt_descriptor *dt, const char *driver)
+{
+	unsigned int i = 0;
+	unsigned int j = 0;
+	char root_name[256] = { };
+	char buffer[512] = { };
+	char *item[128] = { };
+	char *root = NULL;
+	char *p = buffer;
+	int offs = 0;
+
+	offs = fdt_path_offset(dt->blob, driver);
+	if (offs >= 0)
+		return offs;
+
+	strncpy(buffer, driver, sizeof(buffer));
+
+	/* strok modifies the buffer */
+	root = strtok_r(p, "/", &p);
+	if (!root)
+		return -1;
+
+	for (i = 0; i < ARRAY_SIZE(item); i++) {
+		item[i] = strtok_r(p, "/", &p);
+		if (!item[i])
+			break;
+	}
+
+	if (i >= ARRAY_SIZE(item))
+		return -1;
+
+	/* add the root node name */
+	snprintf(root_name, sizeof(root_name), "/%s", root);
+	offs = fdt_path_offset(dt->blob, root_name);
+	if (offs < 0) {
+		offs = add_dt_path_subnode(dt, "/", root);
+		if (offs < 0)
+			return -1;
+	}
+
+	for (j = 0; j < i; j++) {
+		offs = fdt_add_subnode(dt->blob, offs, item[j]);
+		if (offs < 0)
+			return -1;
+	}
+
+	return offs;
+}
+#endif
+
+/*
+ * Guarantee that these drivers will not be used by either EL1 or EL0: set the
+ * device tree node status in the FDT and overlay to disabled and the
+ * secure-status to okay.
+ * Return values:
+ *    0: num_drivers have been reserved (in DTB and optionally in overlay)
+ *   -1: no __symbols__ in DTB, no HEAP left, error creating overlay nodes,
+ *       driver node not found in DTB (but present as a __symbol__),
+ */
+static int reserve_platform_drivers_dt_nodes(struct dt_descriptor *dt)
+{
+	static const char **drivers;
+	static size_t num_drivers;
+	const char * const *drvs = { NULL };
+	const char *path = NULL;
+	unsigned int i = 0;
+	int offs = 0;
+	int len = 0;
+#ifdef CFG_NSEC_DTB_OVERLAY_ADDR
```

Comment:
Please use `if (IS_ENALBED(CFG_NSEC_DTB_OVERLAY_ADDR)` instead of ifdefs where possible.